Mathlib/Analysis/Calculus/FDeriv/Analytic.lean

```diff
@@ -125,8 +217,17 @@ theorem AnalyticOnNhd.iteratedFDeriv [CompleteSpace F] (h : AnalyticOnNhd 𝕜 f
 @[deprecated (since := "2024-09-26")]
 alias AnalyticOn.iteratedFDeriv := AnalyticOnNhd.iteratedFDeriv
 
+lemma AnalyticOnNhd.hasFTaylorSeriesUpToOn [CompleteSpace F]
+    (n : ℕ∞) (h : AnalyticOnNhd 𝕜 f s) :
+    HasFTaylorSeriesUpToOn n f (ftaylorSeries 𝕜 f) s := by
```

Comment:
Am I right that we need `AnalyticOnNhd`, not `AnalyticOn`, to use `ftaylorSeries`, not `ftaylorSeriesWithin`? If yes, then please add a docstring explaining this. Also, could you please explain in the module docstring which theorems need `[CompleteSpace F]` and which work without this assumption?

Reply:
Yes, you're right, I've just explained this. I've also expanded considerably the file-level docstring.